ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -1606,16 +1615,21 @@ int MMInitLayerByType(struct MiraMonVectLayerInfo *hMiraMonLayer)
         }
         if (hMiraMonLayer->MMMap && hMiraMonLayer->MMMap->fMMMap)
         {
+
             hMiraMonLayer->MMMap->nNumberOfLayers++;
             VSIFPrintfL(hMiraMonLayer->MMMap->fMMMap, "[VECTOR_%d]\n",
                         hMiraMonLayer->MMMap->nNumberOfLayers);
+
             VSIFPrintfL(hMiraMonLayer->MMMap->fMMMap, "Fitxer=%s.pnt\n",
-                        CPLGetBasename(hMiraMonLayer->pszSrcLayerName));
+                        pszSrcLayerNameCP1252
+                            ? pszSrcLayerNameCP1252
+                            : CPLGetBasename(hMiraMonLayer->pszSrcLayerName));
         }
 
         if (MMInitPointLayer(hMiraMonLayer))
         {
             // Error specified inside the function
+            CPLFree(pszSrcLayerNameCP1252);
```

Comment:
There are several other return statements in that function where the free is missing.